pkgs/native_toolchain_c/test/helpers.dart

```diff
@@ -170,3 +164,13 @@ Future<String> runOtoolInstallName(Uri libraryUri, String libraryName) async {
       .split(' ')[1];
   return installName;
 }
+
+/// Opens the [DynamicLibrary] at [path] and register a tear down hook to close
+/// it when the current test is done.
+DynamicLibrary openDynamicLibraryForTest(String path) {
+  final library = DynamicLibrary.open(path);
+  // TODO: Remove this ignore when sdk is >=3.1.0.
+  // ignore: sdk_version_since
+  addTearDown(library.close);
```

Comment:
It's fine to just bump the minimum Dart SDK to 3.1

The native assets feature is in experimental and gets fixes etc. So I presume users trying out this feature will likely always be on a very recent SDK.